### ISSUE TITLE
Padded zeroes are not removed in CBOW vector

### ISSUE THREAD

**Issue by luffycodes** [florianmai/word2mat#1]
Is there a specific reason as to why the padded zeroes are added in the CBOW vector? The lookup table learns an embedding for the zeroes and this embedding is added to CBOW vector in case of padded zeroes and missing word.

Code : word2mat.py, line 96
elif self.w2m_type == "cbow":
            cur_emb = torch.sum(word_matrices, 1)

**Comment by florianmai** [comment]
The nn.Embedding class accepts a padding index parameter. The respective embedding is initialized at zero and its gradient is also always set to zero during backpropagation. Hence, the padded zeros should have no impact whatsoever.

**Comment by luffycodes** [comment]
thanks, so cool !
